libcxx/test/std/ranges/range.adaptors/range.stride.view/end.pass.cpp

```diff
@@ -0,0 +1,126 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17, c++20
+
+// constexpr auto end() requires(!__simple_view<_View>)
+// constexpr auto end() const requires(range<const _View>)
+
+// Note: Checks here are augmented by checks in
+// iterator/ctor.copy.pass.cpp.
+
+#include <ranges>
+
+#include "types.h"
+
+template <class T>
+concept HasConstEnd = requires(const T& ct) { ct.end(); };
+
+template <class T>
+concept HasEnd = requires(T& t) { t.end(); };
+
+template <class T>
+concept HasConstAndNonConstEnd =
+    HasConstEnd<T> && requires(T& t, const T& ct) { requires !std::same_as<decltype(t.end()), decltype(ct.end())>; };
+
+template <class T>
+concept HasOnlyNonConstEnd = HasEnd<T> && !HasConstEnd<T>;
+
+template <class T>
+concept HasOnlyConstEnd = HasConstEnd<T> && !HasConstAndNonConstEnd<T>;
+
+struct NoConstView : std::ranges::view_base {
+  int* begin();
+  int* end();
+};
+
+struct UnsimpleConstView : std::ranges::view_base {
+  double* begin();
+  int* begin() const;
+
+  double* end();
+  int* end() const;
+};
+
+struct UnsimpleUnCommonView : std::ranges::view_base {
+  double* begin();
+  char* begin() const;
+
+  void* end();
+  void* end() const;
+};
+
+struct SimpleUnCommonView : std::ranges::view_base {
+  int* begin();
+  int* begin() const;
+
+  void* end();
+  void* end() const;
+};
+
+static_assert(HasOnlyNonConstEnd<std::ranges::stride_view<NoConstView>>);
+static_assert(HasOnlyConstEnd<std::ranges::stride_view<BasicTestView<int*, int*>>>);
+static_assert(HasConstAndNonConstEnd<std::ranges::stride_view<UnsimpleConstView>>);
+
+constexpr bool test_non_default_sentinel() {
+  {
+    LIBCPP_STATIC_ASSERT(std::ranges::__simple_view<BasicTestView<int*, int*>>);
+    static_assert(std::ranges::common_range<BasicTestView<int*, int*>>);
+    static_assert(std::ranges::sized_range<BasicTestView<int*, int*>>);
+    static_assert(std::ranges::forward_range<BasicTestView<int*, int*>>);
+
+    auto v  = BasicTestView<int*, int*>{nullptr, nullptr};
+    auto sv = std::ranges::stride_view<BasicTestView<int*, int*>>(v, 1);
+    static_assert(!std::is_same_v<std::default_sentinel_t, decltype(sv.end())>);
+  }
+
+  {
+    LIBCPP_STATIC_ASSERT(!std::ranges::__simple_view<NoConstView>);
```

Comment:
Our unit tests shouldn't be relying on library implementation details. If there isn't already a test implementation of this concept, please add it.

Reply:
I believe that this is addressed in 48c9f9bbfc6d2010fc9545a9ce61984ccd8ce090.